include/seqan3/core/algorithm/concept.hpp

```diff
@@ -60,10 +89,15 @@ class config_element_base;
 //!\}
 //!\cond
 template <typename config_t>
-SEQAN3_CONCEPT config_element_specialisation = std::semiregular<std::remove_reference_t<config_t>> &&
-requires (config_t c)
+SEQAN3_CONCEPT config_element_specialisation = requires
 {
-    { std::remove_reference_t<config_t>::id };
+    requires std::is_base_of_v<seqan3::pipeable_config_element<config_t>, config_t>;
+    requires std::semiregular<config_t>;
```

Comment:
Why did you omit the `std::remove_reference_t` when testing semi regularity?

Reply:
This concept asks whether a given type is an instance of a pipebale config element. Since this requires semiregularity, it would be wrong to strip away references and cv-qualifiers, because otherwise the caller using a forwarding reference in a function template would think he could call `t{}` which he can't if it is an l-value reference. Hence, it is the responsibility of the caller to ensure that the references are stripped.